tests/component-tests/textToSpeech/js/TextToSpeechTests.js

```diff
@@ -31,113 +31,171 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+    fluid.tests.textToSpeech.testStop = function (tts) {
+        var that = tts;
+        jqUnit.assert("The onStop event should have fired");
+        jqUnit.assertFalse("Should not be speaking", that.model.speaking);
+        jqUnit.assertFalse("Nothing should be pending", that.model.pending);
+        jqUnit.assertFalse("Shouldn't be paused", that.model.paused);
+        jqUnit.assertDeepEq("The queue should be empty", [], that.queue);
+        that.cancel();
+    };
 
-        jqUnit.assertTrue("The Text to Speech component should have initialized", that);
-        jqUnit.assertFalse("Nothing should be speaking", that.model.speaking);
+    fluid.tests.textToSpeech.testPause = function (tts) {
+        var that = tts;
+        jqUnit.assert("The pause event should have fired");
+        jqUnit.assertTrue("Should be speaking", that.model.speaking);
+        jqUnit.assertFalse("Nothing should be pending", that.model.pending);
+        jqUnit.assertTrue("Should be paused", that.model.paused);
+    };
+
+    fluid.tests.textToSpeech.testResume = function (tts) {
+        var that = tts;
+        jqUnit.assert("The resume event should have fired");
+        jqUnit.assertTrue("Should be speaking", that.model.speaking);
         jqUnit.assertFalse("Nothing should be pending", that.model.pending);
         jqUnit.assertFalse("Shouldn't be paused", that.model.paused);
-        jqUnit.start();
-    });
+    };
 
-    fluid.tests.textToSpeech.issueTest("Start and Stop Events", function () {
-        jqUnit.expect(10);
-        var that = fluid.tests.textToSpeech.startStop();
-        that.queueSpeech("Testing start and end events");
-    });
+    fluid.tests.textToSpeech.bypassTest = function () {
+        jqUnit.test("Tests were skipped - browser does not appear to support TTS", function () {
+            jqUnit.assert("TESTS SKIPPED - browser does not support SpeechSynthesis");
+        });
+    };
 
-    // Chrome doesn't properly support pause which causes this test to break.
-    // see: https://code.google.com/p/chromium/issues/detail?id=425553&q=SpeechSynthesis&colspec=ID%20Pri%20M%20Week%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified
-    if (!window.chrome) {
-        fluid.tests.textToSpeech.issueTest("Pause and Resume Events", function () {
-            jqUnit.expect(8);
-            var that = fluid.tests.textToSpeech.pauseResume();
-            that.queueSpeech("Testing pause and resume events");
+    var ttsStatus;
+
+    jqUnit.asyncTest("Confirming if TTS is available", function () {
+        jqUnit.expect(1);
+        var ttsSupport = fluid.textToSpeech.checkTTSSupport();
+        ttsSupport.then(function (){
+            ttsStatus = "TTS is available";
+            fluid.tests.textToSpeech.ttsTestEnvironment();
+        }, function () {
+            ttsStatus = "TTS is unavailable";          
+            fluid.tests.textToSpeech.bypassTest();
         });
-    }
 
+        var ttsTest = function () {
```

Comment:
This is to ensure the actual asynchronous TTS tests have either started to run or been skipped because TTS is not supported in the all-tests context. 

If there's not an asynchronous test wrapper around the promise at lines 184-191, the all-tests runner moves on to the next test (it won't wait for the promise to resolve); this seems to have been the cause of https://issues.fluidproject.org/projects/FLUID/issues/FLUID-5735.

I tried experimenting at first with a `fluid.tests.textToSpeech.ttsTestEnvironment` grade that delayed creation of the `ttsTester` component until the TTS support had been tested for, but couldn't make this work in a way that prevented the all-tests runner from moving on.